src/common/hyperbeam/hb.ts

```diff
@@ -0,0 +1,251 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { JSONValue } from '../../types/common.js';
+import { Logger } from '../logger.js';
+
+export type HBConfig = {
+  url: string;
+  processId: string;
+  logger?: Logger;
+  hbTimeoutMs?: number;
+};
+
+export interface Hyperbeam {
+  meta(): Promise<JSONValue>;
+  now<T extends JSONValue>({
+    path,
+    json,
+  }: {
+    path: string;
+    json?: boolean;
+  }): Promise<T>;
+  compute<T extends JSONValue>({
+    path,
+    json,
+  }: {
+    path: string;
+    json?: boolean;
+  }): Promise<T>;
+  checkHyperBeamCompatibility(params?: { minSlot?: number }): Promise<boolean>;
+}
```

Comment:
nit: we don't need an interface (we won't have multiple implmeentatinos) - so just have a single class with these methods and you can call them from ARIO.

general takeaway: we "over interfaced" early on with ARIO and ANT classes when we didn't need to